c_common/front_end_common_lib/include/spinn_extra.h

```diff
@@ -0,0 +1,1439 @@
+        uint int1 : 1;          //!< External interrupt request 1
+        uint gpio8 : 1;         //!< Signal on GPIO[8]
+        uint gpio9 : 1;         //!< Signal on GPIO[9]
+    };
+    uint value;
+} vic_mask_t;
+
+//! VIC registers
+typedef struct {
+    const vic_mask_t irq_status;    //!< IRQ status register
+    const vic_mask_t fiq_status;    //!< FIQ status register
+    const vic_mask_t raw_status;    //!< raw interrupt status register
+    vic_mask_t int_select;          //!< interrupt select register
+    vic_mask_t int_enable;          //!< interrupt enable set register
+    vic_mask_t int_disable;         //!< interrupt enable clear register
+    vic_mask_t soft_int_enable;     //!< soft interrupt set register
+    vic_mask_t soft_int_disable;    //!< soft interrupt clear register
+    bool protection;                //!< protection register
+    const uint _padding[3];
+    //! current vector address register
+    vic_interrupt_handler_t vector_address;
+    //! default vector address register
+    vic_interrupt_handler_t default_vector_address;
+} vic_control_t;
+
+//! VIC individual vector control
+typedef struct {
+    uint source : 5;                //!< interrupt source
+    uint enable : 1;                //!< interrupt enable
+    uint : 26;
+} vic_vector_control_t;
+
+ASSERT_WORD_SIZED(vic_mask_t);
+ASSERT_WORD_SIZED(vic_interrupt_handler_t);
+ASSERT_WORD_SIZED(vic_vector_control_t);
+
+//! VIC registers
+static volatile vic_control_t *const vic_control =
+        (vic_control_t *) VIC_BASE_UNBUF; // NB unbuffered!
+//! VIC interrupt handlers. Array of 32 elements.
+static volatile vic_interrupt_handler_t *const vic_interrupt_vector =
+        (vic_interrupt_handler_t *) (VIC_BASE + 0x100);
+//! VIC individual interrupt control. Array of 32 elements.
+static volatile vic_vector_control_t *const vic_interrupt_control =
+        (vic_vector_control_t *) (VIC_BASE + 0x200);
+
+//! \}
+
+// ---------------------------------------------------------------------
+//! \name 6. Counter/Timer
+//! \brief Every core has a pair of counters/timers.
+//! \{
+
+//! Timer control register
+typedef struct {
+    uint one_shot : 1;          //!< 0 = wrapping mode, 1 = one shot
+    uint size : 1;              //!< 0 = 16 bit, 1 = 32 bit
+    uint pre_divide : 2;        //!< divide input clock (see ::timer_pre_divide)
+    uint : 1;
+    uint interrupt_enable : 1;  //!< enable interrupt (1 = enabled)
+    uint periodic_mode : 1;     //!< 0 = free-running; 1 = periodic
+    uint enable : 1;            //!< enable counter/timer (1 = enabled)
+    uint : 24;
+} timer_control_t;
+
+//! Values for ::timer_control_t::pre_divide
+enum timer_pre_divide {
+    TIMER_PRE_DIVIDE_1 = 0,     //!< Divide by 1
+    TIMER_PRE_DIVIDE_16 = 1,    //!< Divide by 16
+    TIMER_PRE_DIVIDE_256 = 2    //!< Divide by 256
+};
+
+//! Timer interrupt status flag
+typedef struct {
+    uint status : 1;            //!< The flag bit
+    uint : 31;
```

Comment:
comment and named constant?

Reply:
Commented as `padding`. Will not name the constant; it's literally just 31 bits of padding that should be ignored.